onnxruntime/core/session/inference_session.cc

```diff
@@ -1655,6 +1656,22 @@ common::Status InferenceSession::Initialize() {
     const Env& env = Env::Default();
     env.GetTelemetryProvider().LogSessionCreationStart();
 
+    const std::unordered_map<std::string, std::unique_ptr<OrtExecutionProviderFactory>>& custom_ep_factories = environment_.GetCustomEpFactories();
+    if (custom_ep_factories.size() > 0) {
```

Comment:
check ep_name in the session in order to check the custom ep in the session only